### DISCUSSION TITLE
HTML in error messages

### DISCUSSION THREAD

**Discussion by roypearce** [react-hook-form/react-hook-form#3654] [Ideas]
I see based on the generated Types that error messages from a `pattern` object or the return message from a `validate` function for example can only be strings. Would it be possible for an error message to be a `React.ReactNode` instead of just a `string`? 

It would be helpful to be able to have an error message that also contains a link to documentation as one of the use cases, or be able to emphasize a word or phrase.

**Comment by bluebill1049** [comment]
currently we only support this feature with JS, not with TS, it does create type issue with validation function

you will have to combine with JSX and error message

**Reply by roypearce** [reply]
Yes, we are using typescript so that is why this is an issue. 

Are you saying that changing the message type to React.ReactNode would not be possible right now due to an existing issue? If so, is there something I can do to help resolve it and make that happen?

**Reply by bluebill1049** [reply]
yes, there is an issue with typing React.ReactNode. I can't remember on top of my head, feel free to fork the repo and have a play when you free.
